cli_helpers/tabular_output/sql_output_adapter.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+
+supported_formats = (
+    "sql-insert",
+    "sql-update",
+    "sql-update-1",
+    "sql-update-2",
+)
+
+preprocessors = ()
+
+
+def escape_for_sql_statement(value):
+    if isinstance(value, bytes):
+        return f"X'{value.hex()}'"
+    else:
+        return "'{}'".format(value)
+
+
+def adapter(data, headers, table_format=None, **kwargs):
+    """
+    This function registers supported_formats to default TabularOutputFormatter
+
+    Parameters:
+        data: query result
+        headers: columns
+        table_format: values from supported_formats
+        kwargs:
+            extract_tables: extract_tables function. For example, in pgcli.packages.parseutils.tables there is a function extract_tables
+            delimiter: Character surrounds table name or column name when it conflicts with sql keywords.
+                       For example, mysql uses ` and postgres uses "
+    """
+    extract_table_func = kwargs.get("extract_tables")
+    if not extract_table_func:
+        raise ValueError("extract_tables function should be registered first")
+
+    tables = extract_table_func(formatter.query)
+    delimiter = kwargs.get("delimiter")
+    if not isinstance(delimiter, str):
+        delimiter = '"'
+
+    if tables is not None and len(tables) > 0:
```

Comment:
This adapter doesn't really make sense for more than one table, correct? Perhaps there should be an error if we have more than one table.

Reply:
Yes, this formatter is convenient is generate INSERTION SQL or UPDATING SQL relates to one table. In those cases when SQL with multiple tables is run, usually queried results earn more concern than considering to importing them. So I thought it was ok to use a fake table name "DUAL" to indicate this situation, and you?

Reply:
I think we should raise an exception in this case.